src/mame/video/k057714.cpp

```diff
@@ -462,15 +503,15 @@ void k057714_device::draw_frame(int frame, bitmap_ind16 &bitmap, const rectangle
 
 int k057714_device::draw(screen_device &screen, bitmap_ind16 &bitmap, const rectangle &cliprect)
 {
-	if (m_display_width != 0 && m_display_height != 0)
+	if (m_display_is_dirty)
 	{
-		rectangle visarea = screen.visible_area();
-		if (visarea.max_x != m_display_width || visarea.max_y != m_display_height)
-		{
-			visarea.max_x = m_display_width;
-			visarea.max_y = m_display_height;
-			screen.configure(m_display_width, m_display_height, visarea, screen.frame_period().attoseconds());
-		}
+		auto htotal = m_display_h_visarea + m_display_h_frontporch + m_display_h_backporch + m_display_h_syncpulse;
+		auto vtotal = m_display_v_visarea + m_display_v_frontporch + m_display_v_backporch + m_display_v_syncpulse;
+
+		rectangle visarea(0, m_display_h_visarea - 1, 0, m_display_v_visarea - 1);
+		screen.configure(htotal, vtotal, visarea, HZ_TO_ATTOSECONDS(screen.clock()) * htotal * vtotal);
+
+		m_display_is_dirty = false;
```

Comment:
Please consider moving this block to a specific `inline void crtc_set_params` function, namely because:

1. for performance: it mostly branch false on every single draw call;
2. for a code smell: I believe that `const cliprect` isn't updated at this point, if the successive drawing functions don't honor your update then it can possibly cause a bitmap overflow crash;
3. for extensibility: attached screen definitely go off sync when HW changes vertical parameter(s) until next frame occurs, something that may eventually need a one shot timer in this device chain;

Reply:
Some of those things I was kind of worried about but wasn't sure how to do it better. I rewrote that part of the code now, thanks.